engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -893,4 +904,59 @@ public Map<String, String> renameMap(final WhereFilter filter, final ColumnSourc
     public PushdownFilterContext makePushdownFilterContext() {
         return new BasePushdownFilterContext();
     }
+
+    public static final class Builder implements Closeable {
+
+        private final PushdownResult[] results;
+
+        private long matchCount;
+        private long maybeMatchCount;
+        private long notMatchCount;
+
+        private Builder(int numResults) {
+            results = new PushdownResult[numResults];
+            matchCount = 0;
+            maybeMatchCount = 0;
+            notMatchCount = 0;
+        }
+
+        public void add(final int ix, final long selectionSize, final PushdownResult results) {
```

Comment:
I think a javadoc comment explaining the transfer of ownership of `results` would be useful here.